scipy/stats/_quantile.py

```diff
@@ -257,3 +295,43 @@ def quantile(x, p, *, method='linear', axis=0, nan_policy='propagate', keepdims=
         res = xp.squeeze(res, axis=axis)
 
     return res[()] if res.ndim == 0 else res
+
+
+def _quantile_hf(y, p, n, method, xp):
+    ms = dict(inverted_cdf=0, averaged_inverted_cdf=0, closest_observation=-0.5,
+              interpolated_inverted_cdf=0, hazen=0.5, weibull=p, linear=1 - p,
+              median_unbiased=p / 3 + 1 / 3, normal_unbiased=p / 4 + 3 / 8)
+    m = ms[method]
+    jg = p*n + m - 1
+    j = jg // 1
+    g = jg % 1
+    if method == 'inverted_cdf':
+        g = xp.astype((g > 0), jg.dtype)
+    elif method == 'averaged_inverted_cdf':
+        g = (1 + xp.astype((g > 0), jg.dtype)) / 2
+    elif method == 'closest_observation':
+        g = (1 - xp.astype((g == 0) & (j % 2 == 1), jg.dtype))
+    if method in {'inverted_cdf', 'averaged_inverted_cdf', 'closest_observation'}:
+        g = xp.asarray(g)
+        g[jg < 0] = 0
+    j = xp.clip(j, 0., n - 1)
+    jp1 = xp.clip(j + 1, 0., n - 1)
+    j = xp.astype(j, xp.int64)
+    jp1 = xp.astype(jp1, xp.int64)
+
+    return ((1 - g) * xp_take_along_axis(y, j, axis=-1)
+            + g * xp_take_along_axis(y, jp1, axis=-1))
+
+
+def _quantile_hd(y, p, n, xp):
+    p = xp.moveaxis(p, -1, 0)[..., np.newaxis]
+    a = p * (n + 1)
+    b = (1 - p) * (n + 1)
+    i = xp.arange(y.shape[-1] + 1)
+    i = xp.astype(i, y.dtype)
+    # There will be edge case bugs with a == 0 or b == 0 due to gh-8411
+    w = betainc(a, b, i / n)
+    w = w[..., 1:] - w[..., :-1]
+    w[xp.isnan(w)] = 0
```

Comment:
`xp.where` to avoid boolean indexing?

Reply:
I'll use `xpx.at.set`. I thought I did...